transport/tchannel/handler.go

```diff
@@ -262,6 +264,20 @@ func (hw *handlerWriter) SetApplicationError() {
 	hw.applicationError = true
 }
 
+func (hw *handlerWriter) SetApplicationFailure(applicationFailure *transport.ApplicationFailure) {
```

Comment:
Note that we're missing the outbound side of the TChannel changes. We need to reconstruct the `ApplicationFailure` on the client side, so that outbound middleware, (ie observability/user middleware) can read it and also emit metrics/logs correctly.

Reply:
Yes, that is the next step. Shall we take it as another PR to avoid having this one becoming even bigger?